src/TorchSharp/NN/Module.cs

```diff
@@ -565,7 +565,15 @@ protected CustomModule(string name, params parameter.Parameter[] parameters) : b
                     var pparray = paramsPinned.CreateArray(@params);
                     var gparray = wGradPinned.CreateArray(withGrads);
 
-                    ForwardFunctionC forwardNative = t => (forward(new Tensor(t)).Handle);
+                    ForwardFunctionC forwardNative = t => {
+                        var input = new Tensor(t);
+                        var output = forward(input);
+                        // handles must live on - we don't own them
+                        GC.SuppressFinalize(output);
```

Comment:
If we need absolutely correct behavior, output tensor might need to be cloned. This would be required if two custom modules cache and share the same instance of `Tensor` and simply return it in `forward`. In this case we would give `Handle` away twice, but the second time `SuppressFinalize` won't actually adjust ownership.